controllers/nodehealthcheck_controller.go

```diff
@@ -341,7 +332,21 @@ func (r *NodeHealthCheckReconciler) remediate(node *v1.Node, nhc *remediationv1a
 		}
 	}
 
-	// set control plane marker label
+	currentTemplate, timeout, err := rm.GetCurrentTemplateWithTimeout(node, nhc)
+	if err != nil {
+		if _, ok := err.(resources.NoTemplateLeftError); ok {
+			log.Error(err, "Remediation timed out, and no template left to try")
+			r.Recorder.Event(nhc, eventTypeWarning, eventReasonNoTemplateLeft, fmt.Sprintf("Remediation timed out, and no template left to try. %s", err.Error()))
+			// there is nothing we can do about this
+			return nil, nil
```

Comment:
IIUC, this means that we will attempt remediation on the next node even though we didn't successfully remediated the current node.
I'm not sure that was the original intention.

Reply:
Why shouldn't we remediate the next node? 🤔

Reply:
We are not in `Reconcile` here, just in `remediate` of a single node. `nil, nil` means no need for a scheduled Requeue, and no error happened, for this node.